src/Components/Facility/PatientNotesDetailedView.tsx

```diff
@@ -0,0 +1,173 @@
+import { useEffect, useState } from "react";
+import CircularProgress from "../Common/components/CircularProgress";
+import routes from "../../Redux/api";
+import { PaitentNotesReplyModel, PatientNotesModel } from "./models";
+import request from "../../Utils/request/request";
+import PatientNoteCard from "./PatientNoteCard";
+import RichTextEditor from "../Common/RichTextEditor/RichTextEditor";
+import * as Notification from "../../../src/Utils/Notifications";
+import CareIcon from "../../CAREUI/icons/CareIcon";
+import DoctorNoteReplyPreviewCard from "./DoctorNoteReplyPreviewCard";
+
+interface Props {
+  patientId: string;
+  consultationId: string;
+  noteId: string;
+  thread: PatientNotesModel["thread"];
+  setThreadViewNote?: (note: string) => void;
+}
+
+const PatientNotesDetailedView = (props: Props) => {
+  const { patientId, consultationId, noteId, thread, setThreadViewNote } =
+    props;
+  const [isLoading, setIsLoading] = useState(true);
+  const [reload, setReload] = useState(false);
+  const [state, setState] = useState<PatientNotesModel>();
+  const [noteField, setNoteField] = useState("");
+  const [reply_to, setReplyTo] = useState<PaitentNotesReplyModel | undefined>(
+    undefined,
+  );
+
+  const onAddNote = async () => {
+    if (!/\S+/.test(noteField)) {
+      Notification.Error({
+        msg: "Note Should Contain At Least 1 Character",
+      });
+      return;
+    }
+
+    const { res, data } = await request(routes.addPatientNote, {
+      pathParams: {
+        patientId: patientId,
+      },
+      body: {
+        note: noteField,
+        thread,
+        consultation: consultationId,
+        reply_to: noteId,
+      },
+    });
+
+    if (res?.status === 201) {
+      Notification.Success({ msg: "Note added successfully" });
+      setNoteField("");
+    }
+
+    return data?.id;
+  };
+  const fetchNotes = async () => {
+    setIsLoading(true);
+
+    const { data } = await request(routes.getPatientNote, {
+      pathParams: {
+        patientId: patientId,
+        noteId,
+      },
+      query: {
+        consultation: consultationId,
+        thread,
+      },
+    });
+
+    if (data) {
+      setState(data);
+    }
+    setIsLoading(false);
+    setReload?.(false);
+  };
+
+  useEffect(() => {
+    if (reload) {
+      fetchNotes();
+    }
+  }, [reload]);
+
+  useEffect(() => {
+    fetchNotes();
+  }, [thread, noteId]);
+
+  useEffect(() => {
+    setReload?.(true);
+  }, []);
```

Comment:
⚠️ unreadable / hard to understand. comments might help

Reply:
updated with comments